tekton/predicates_test.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2022 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tekton
+
+import (
+	"context"
+	"reflect"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Predicates", func() {
+
+	const (
+		pipelineRunPrefixName = "test-pipeline"
+		apiVersion            = "appstudio.redhat.com/v1alpha1"
+		namespace             = "default"
+	)
+
+	var release *v1alpha1.Release
+	var releasePipelineRun *ReleasePipelineRun
+
+	BeforeEach(func() {
+
+		release = &v1alpha1.Release{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: apiVersion,
+				Kind:       "Release",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "testrelease-",
+				Namespace:    namespace,
+			},
+			Spec: v1alpha1.ReleaseSpec{
+				ApplicationSnapshot: "testsnapshot",
+				ReleaseLink:         "testreleaselink",
+			},
+		}
+		ctx := context.Background()
+
+		// The code below sets the ownership for the Release Object
+		kind := reflect.TypeOf(v1alpha1.Release{}).Name()
+		gvk := v1alpha1.GroupVersion.WithKind(kind)
+		controllerRef := metav1.NewControllerRef(release, gvk)
+
+		// Creating a release
+		Expect(k8sClient.Create(ctx, release)).Should(Succeed())
+		release.SetOwnerReferences([]metav1.OwnerReference{*controllerRef})
+
+		// Need to set the Kind and APIVersion as it loses it due to:
+		// https://github.com/kubernetes-sigs/controller-runtime/issues/1870
+		release.TypeMeta.APIVersion = apiVersion
+		release.TypeMeta.Kind = "Release"
+
+		// Creates the PipelineRun Object
+		releasePipelineRun = NewReleasePipelineRun(pipelineRunPrefixName, namespace)
+		Expect(k8sClient.Create(ctx, releasePipelineRun.AsPipelineRun())).Should(Succeed())
+	})
+
+	AfterEach(func() {
+		_ = k8sClient.Delete(ctx, release)
+		_ = k8sClient.Delete(ctx, releasePipelineRun.AsPipelineRun())
+	})
+
+	Context("Testing ReleasePipelineRunSucceededPredicate", func() {
```

Comment:
I think that the descriptions in the tests should be a bit more verbose. For example, in this one I'd write something like:
```suggestion
	Context("when using the ReleasePipelineRunSucceededPredicate predicate", func() {
```
And then the tests would be something like:
```
It("should ignore creation events")
It("should ignore deletion events")
It("should ignore generic events")
It("returns true when an update event is received for a succeeded release PipelineRun")
```

Reply:
Ack'ing here for all the comments with the same context. I will changes these labels.